api/src/main/java/org/openmrs/module/openconceptlab/UpdateService.java

```diff
@@ -284,4 +286,19 @@ public void scheduleUpdate() {
 		items.setMaxResults(max);
 		return items.list();
 	}
+
+	/**
+	 * @param update the update to be passed
+	 * @param  states set of states passed
+	 * @return a count of items
+	 */
+	public Integer getUpdateItemsCount(Update update, Set<State> states) {
+		Criteria items = getSession().createCriteria(Item.class);
+		items.add(Restrictions.eq("update", update));
+		items.addOrder(Order.desc("state"));
+		if (!(states.isEmpty())) {
+			items.add(Restrictions.in("state", states));
+		}
+		return  ((Long) items.setProjection(Projections.rowCount()).uniqueResult()).intValue();

```

Comment:
It's safer to cast it to Number instead of Long. In some older OpenMRS versions it is Integer not Long.